lib/plugins/deploy.js

```diff
@@ -39,12 +39,11 @@ class Deploy {
           throw new ServerlessError(errorMessage);
         }
         if (this.options.function) {
-          // If the user has given a function parameter, spawn a function deploy
-          // and terminate execution right afterwards. We did not enter the
-          // deploy lifecycle yet, so nothing has to be cleaned up.
-          await this.serverless.pluginManager.spawn('deploy:function', {
-            terminateLifecycleAfterExecution: true,
-          });
+          this.serverless.logDeprecation(
+            'CLI_DEPLOY_FUNCTION_OPTION',
+            'The `--function` or `-f` option for `deploy` command has been r' +
+              'emoved. Please use `sls deploy function` command instead.'
+          );
```

Comment:
This PR should only implement a deprecation notice, without removing the functionality.